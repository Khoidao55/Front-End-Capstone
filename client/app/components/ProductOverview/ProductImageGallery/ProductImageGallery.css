.image-container {
  display: flex;
  margin-left: 50px;
}

.image-right-btn-active {
  display: flex;
  position: absolute;
  margin-left: 630px;
  margin-top: 337px;
}

.image-right-btn-disabled {
  display: none;
}

.image-left-btn-active {
  display: flex;
  position: absolute;
  margin-left: 70px;
  margin-top: 337px;
}

.image-left-btn-disabled {
  display: none;
}

.modal {
  position: fixed; /* Stay in place */
  z-index: 20; /* Sit on top */
  padding-top: 5px; /* Location of the box */
  left: 0;
  top: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  overflow: auto; /* Enable scroll if needed */
  background-color: rgb(0,0,0); /* Fallback color */
  background-color: rgba(0,0,0,0.9); /* Black w/ opacity */
}