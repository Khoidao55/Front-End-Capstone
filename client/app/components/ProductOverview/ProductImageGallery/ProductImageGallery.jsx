import React, { useState } from 'react';
import propTypes from 'proptypes';
import DefaultView from './DefaultView/DefaultView.jsx';
import ExpandedView from './ExpandedView/ExpandedView.jsx';
import ProductThumbnailScroll from './ProductThumbnailScroll/ProductThumbnailScroll.jsx';
import './ProductImageGallery.css';

const ProductImageGallery = ({ style }) => {
  if (style !== '') {
    const toggle = false;
    const [currentPhoto, setCurrentPhoto] = useState(0);
    const [expandView, setExpandView] = useState(toggle);

    const onClickRightChange = () => {
      console.log('clicked right');
      if (currentPhoto !== style.photos.length - 1) {
        setCurrentPhoto(currentPhoto + 1);
      }
    };

    const onClickLeftChange = () => {
      console.log('clicked left');
      if (currentPhoto > 0) {
        setCurrentPhoto(currentPhoto - 1);
      }
    };
    const onClickZoom = () => {
      console.log('clicked on image', toggle);
      setExpandView((view) => !view);
    };

    return (
      <div>
        <div className="product-thumbnail"><ProductThumbnailScroll arrayOfPhoto={style} /></div>
        {
        expandView === false
          ? (
            <div className="image-container">
              <i className={`image-left-btn-${currentPhoto !== 0 ? `active` : `disabled`} fas fa-chevron-left`} onClick={onClickLeftChange} alt="Left Button"></i>
              <DefaultView
                onClickImage={onClickZoom}
                defaultPhoto={style.photos[currentPhoto]}
              />
                <i className={`image-right-btn-${currentPhoto !== (style.photos.length - 1) ? `active` : `disabled`} fas fa-chevron-right`} onClick={onClickRightChange}alt="Right Button"></i>
            </div>
          )
          : (
            <div onClickImage={onClickZoom} className="modal">
              <ExpandedView
                onClickImage={onClickZoom}
                defaultPhoto={style.photos[currentPhoto]}
              />
            </div>
          )
        }
      </div>
    );
  }
  return (
    <div />
  );
};

ProductImageGallery.propTypes = {
  style: propTypes.object.isRequired,
};

export default ProductImageGallery;
