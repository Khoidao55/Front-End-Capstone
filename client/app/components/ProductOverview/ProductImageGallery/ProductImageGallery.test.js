import React from 'react';
import { shallow } from 'enzyme';
import ProductImageGallery from './ProductImageGallery.jsx';

describe('Product Image Gallery', () => {
  const dummyProps = {
    expandView: false,
    currentPhoto: 0,
    onClickZoom: () => { console.log('zoom'); },
    onClickChangeThumbnail: () => { console.log('thumbnail'); },
    onClickRightChange: () => { console.log('right click'); },
    style: {
      photos: [{}, {}, {}],
    },
  };
  test('should render my component', () => {
    const wrapper = shallow(<ProductImageGallery {...dummyProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
