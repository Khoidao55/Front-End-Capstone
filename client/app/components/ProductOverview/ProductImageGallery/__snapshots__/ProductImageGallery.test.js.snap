// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image Gallery should render my component 1`] = `ShallowWrapper {}`;
